test_runner/regress/test_tenants.py

```diff
@@ -201,3 +202,39 @@ def get_ps_metric_samples_for_tenant(tenant_id: TenantId) -> List[Sample]:
 
         post_detach_samples = set([x.name for x in get_ps_metric_samples_for_tenant(tenant)])
         assert post_detach_samples == set()
+
+
+def test_pageserver_with_empty_tenants(neon_simple_env: NeonEnv):
+    env = neon_simple_env
+    client = env.pageserver.http_client()
+
+    tenant_without_timelines_dir = env.initial_tenant
+    shutil.rmtree(Path(env.repo_dir) / "tenants" / str(tenant_without_timelines_dir) / "timelines")
```

Comment:
That should never happen in normal pageserver and we manually remove the dir to check the failure scenario, right?
Please, add a comment.

Reply:
This is actually another good case where I think I'm doing wrong.
Right now, such tenant will be omitted, since we try to collect its timelines and fail, on `fn collect_timelines_for_tenant` level, completely ignoring such tenants on loading, not putting it into memory (as the test shows).

Later, if somebody attempts to create such tenant, it will error, because there are some files on the local disk.
pageserver does not remove such files and does not show anything in the management HTTP API, which I think is wrong.

I would love to change it more later, to actually return such tenants with "Broken" tenant state. I'm not sure if it's worth doing now, wdyt?

Reply:
I think it would be nice to track such tenants somehow.
to help future us who will debug obscure "this tenant already exists" errors.
It's not urgent though. Feel free to create a separate issue

Reply:
https://github.com/neondatabase/neon/issues/2552